app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -0,0 +1,136 @@
+  This snippet integrates into the existing "redhat_register" snippet for efficient host registration.
+-%>
+
+# Define the path to rhsm.conf
+RHSM_CFG=/etc/rhsm/rhsm.conf
+
+# Backup the original rhsm.conf file
+if [ -f $RHSM_CFG ] ; then
+  test -f $RHSM_CFG.bak || cp $RHSM_CFG $RHSM_CFG.bak
+fi
+
+<% if plugin_present?('katello') -%>
+  # Define the path to the Katello server CA certificate
+  KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.pem
+
+  # If SSL_CA_CERT is not set, create a temporary file for it
+  if [ -z "$SSL_CA_CERT" ]; then
+    SSL_CA_CERT=$(mktemp)
+    cat << EOF > "$SSL_CA_CERT"
+<%= foreman_server_ca_cert %>
+EOF
+  fi
+
+  <% if @registration_method == 'registration' -%>
+    # rhn-client-tools conflicts with subscription-manager package
+    # since rhn tools replaces subscription-manager, we need to explicitly
+    # install subscription-manager after the rhn tools cleanup
+    if [ x$ID = xol ]; then
+      $PKG_MANAGER_REMOVE rhn-client-tools
+      $PKG_MANAGER_INSTALL --setopt=obsoletes=0 subscription-manager
+    fi
+
+    <% if truthy?(@force) -%>
+      # Unregister host and remove all local system and subscription data
+      if [ -x "$(command -v subscription-manager)" ] ; then
+        subscription-manager unregister || true
+        subscription-manager clean
+      fi
+
+      $PKG_MANAGER_REMOVE katello-ca-consumer\*
+    <% end -%>
+  <% end -%>
+
+  # Prepare the SSL certificate
+  mkdir -p /etc/rhsm/ca
+  cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+  chmod 644 $KATELLO_SERVER_CA_CERT
+<% end -%>
+
+# Prepare subscription-manager
+if ! [ -x "$(command -v subscription-manager)" ] ; then
+  $PKG_MANAGER_INSTALL subscription-manager
+else
+  <% if @registration_method == 'registration' -%>
+    $PKG_MANAGER_UPGRADE subscription-manager
+  <% end %>
+fi
+
+# Check if rhsm.conf exists
+if ! [ -f $RHSM_CFG ] ; then
+  echo "'$RHSM_CFG' not found, cannot configure subscription-manager"
+  <% if plugin_present?('katello') -%>
+    rm -f $SSL_CA_CERT
+  <% end -%>
+  exit 1
+fi
+
+<% if @registration_method == 'registration'
+     if plugin_present?('katello')
+       server_hostname = @rhsm_url.host if @rhsm_url
+       server_port = @rhsm_url.port if @rhsm_url
+       server_prefix = @rhsm_url.path if @rhsm_url
+       repo_ca_cert = "$KATELLO_SERVER_CA_CERT"
+       rhsm_baseurl = @pulp_content_url
+     end
+   elsif @registration_method == 'provisioning'
+     if plugin_present?('katello')
+       server_hostname = @host.content_source
+       server_port = "443"
+       server_prefix = "/rhsm"
+       repo_ca_cert = "$KATELLO_SERVER_CA_CERT"
+       rhsm_baseurl = "https://#{@host.content_source}/pulp/content/"
+     else
+       server_hostname = "subscription.rhsm.redhat.com"
+       server_port = "443"
+       server_prefix = "/subscription"
+       repo_ca_cert = "/etc/rhsm/ca/redhat-uep.pem"
+       rhsm_baseurl = "https://cdn.redhat.com"
+     end
+   end
+%>
```

Comment:
This code block still requires more work. My question is if we need the last part, beginning from line 90 to 95.

Reply:
```ruby
     if plugin_present?('katello')
       server_hostname = @host.content_source
       server_port = "443"
       server_prefix = "/rhsm"
       repo_ca_cert = "$KATELLO_SERVER_CA_CERT"
       rhsm_baseurl = "https://#{@host.content_source}/pulp/content/"
```

port, prefix and baseurl should be delivered from the `@host.content_source` object

Reply:
Thanks, I fixed it: https://github.com/theforeman/foreman/pull/9837/files#diff-f75ade9741dfd87c0aa7737555a920b2f94391d5b1ed010af6e7e3faaaf2c40dR89-R93